psydac/api/ast/parser.py

```diff
@@ -997,10 +987,10 @@ def _visit_ComputeKernelExpr(self, expr, op=None, lhs=None, **kwargs):
             ext     = target.ext if isinstance(target, Boundary) else 1
 
             J       = LogicalExpr(Jacobian(mapping), mapping=mapping, dim=dim, subs=True)
-            J       = SymbolicExpr(J)
-            values  = [ext * J.cofactor(i, j=axis) for i in range(J.shape[0])]
-            normalization = sum(v**2 for v in values)**0.5
-            values  = [v.simplify() for v in values]
+            J       = SymbolicExpr(J).inv()
+            values  = ext * J[axis, :]
+            normalization = values.dot(values)**0.5
+            values  = [v for v in values]
```

Comment:
It appears that the original algorithm had a bug, and it was missing the determinant factor.

But I am not convinced by these changes. The previous version was explicitly computing one row of the inverse Jacobian matrix; instead you are now computing the whole inverse and then extracting the row, which is significantly slower.

Reply:
yes it can be improved we actually have access to the inverse now by the new modifications in Sympde

Reply:
OK, let's fix it in this PR.